packages/gatsby/src/utils/api-node-docs.ts

```diff
@@ -133,6 +133,16 @@ export const sourceNodes = true
  */
 export const onCreateNode = true
 
+/**
+ * Called when a new node is created before the `onCreateNode` handler is called.
+ * This is an optimization that can prevent the `onCreateNode` handler to be scheduled
+ * if the plugin already knows it's not interested in processing this node.
+ *
+ * @example
+ * exports.onCreateNodeSyncTest = (node) => node.internal.type === 'Image'
+ */
+export const onCreateNodeSyncTest = true
```

Comment:
Can we rename the API to `shouldCreateNode`, `shouldCreateNodePredicate` or `shouldCreateNodeTest`

```suggestion
/**
 * Called when a new node is created before the `onCreateNode` handler is called.
 * This is an optimization that can prevent the `onCreateNode` handler to be scheduled
 * if the plugin already knows it's not interested in processing this node.
 *
 * @example
 * exports.onCreateNodeSyncTest = (node) => node.internal.type === 'Image'
 */
export const shouldCreateNode = true
```

Reply:
Sure

Reply:
Let's go with `shouldOnCreateNode`, then.